python-avocado.spec

```diff
@@ -1,8 +1,24 @@
 %global srcname avocado
+
+# Conditional for release vs. snapshot builds. Set to 1 for release build.
+%if ! 0%{?rel_build:1}
+%global rel_build 1
+%endif
+
+# Settings used for build from snapshots.
+%if 0%{?rel_build}
+%global gittar		%{srcname}-%{version}.tar.gz
+%else
 %if ! 0%{?commit:1}
- %define commit 1d717446d28ff852fe5de895f14c7985c63ada3e
+%global commit		817c25c2bbedb6d43e5fa2ad0a78698f1416ab49
+%endif
+%if ! 0%{?commit_date:1}
+%global commit_date	20170407
+%endif
+%global shortcommit	%(c=%{commit};echo ${c:0:7})
+%global gitrel		.%{commit_date}git%{shortcommit}
+%global gittar		%{srcname}-%{shortcommit}.tar.gz
```

Comment:
This is really confusing, but after re-arranging it looks like it's correct... How about:

```diff
diff --git a/python-avocado.spec b/python-avocado.spec
index 82c49ee..78cb6bc 100644
--- a/python-avocado.spec
+++ b/python-avocado.spec
@@ -7,17 +7,17 @@
 
 # Settings used for build from snapshots.
 %if 0%{?rel_build}
-%global gittar         %{srcname}-%{version}.tar.gz
+       %global gittar          %{srcname}-%{version}.tar.gz
 %else
-%if ! 0%{?commit:1}
-%global commit         817c25c2bbedb6d43e5fa2ad0a78698f1416ab49
-%endif
-%if ! 0%{?commit_date:1}
-%global commit_date    20170407
-%endif
-%global shortcommit    %(c=%{commit};echo ${c:0:7})
-%global gitrel         .%{commit_date}git%{shortcommit}
-%global gittar         %{srcname}-%{shortcommit}.tar.gz
+       %if ! 0%{?commit:1}
+               %global commit          817c25c2bbedb6d43e5fa2ad0a78698f1416ab49
+       %endif
+       %if ! 0%{?commit_date:1}
+               %global commit_date     20170407
+       %endif
+       %global shortcommit     %(c=%{commit};echo ${c:0:7})
+       %global gitrel          .%{commit_date}git%{shortcommit}
+       %global gittar          %{srcname}-%{shortcommit}.tar.gz
 %endif
 
 # selftests are provided but may need to skipped because many of
```

Anyway if I understand this correctly (not an RPM expert) the main difference is that the released version will use the `tag` instead of `sha` as a pointer to the sources and some other name-related changes?